project/UI/Carbon/Carbon.swift

```diff
@@ -2,71 +2,70 @@
-                                     .init(name: CarbonCommand.theme.rawValue, placeholder: "", description: "carbon's theme.", default: "dracula"),
-                                     .init(name: CarbonCommand.size.rawValue, placeholder: "", description: "export file size [1-5].", default: "2"),
-                                     .init(name: CarbonCommand.font.rawValue, placeholder: "", description: "carbon's font type.", default: "fira-code"),
-                                     .init(name: CarbonCommand.lines.rawValue, placeholder: "", description: "shows/hides lines of code [true | false].", default: "true"),
-                                     .init(name: CarbonCommand.watermark.rawValue, placeholder: "", description: "shows/hides the watermark [true | false].", default: "true"))
+    @ArgumentParser.Option(default: .x2, help: "export file size [1-5]")
+    var size: CarbonStyle.Size
     
-    return arguments(console: console)
-        .flatMap { (input, output, style) in
-            nef.Carbon.render(playgroundsAt: input, style: style, into: output)
-                      .provide(console)^
-                      .mapError { _ in .render() }
-                      .foldM({ _ in console.exit(failure: "rendering Xcode Playgrounds from '\(input.path)'") },
-                             { _ in console.exit(success: "rendered Xcode Playgrounds in '\(output.path)'")   }) }^
-        .reportStatus(in: console)
-        .foldM({ e in console.exit(failure: "\(e)")        },
-               { success in console.exit(success: success) })
-        .unsafeRunSyncEither()
+    @ArgumentParser.Option(default: .firaCode, help: "Carbon font type")
+    var font: CarbonStyle.Font
+    
+    @ArgumentParser.Option(name: .customLong("show-lines"), default: true, help: "Shows/hides lines of code [true | false]")
+    var lines: Bool
+    
+    @ArgumentParser.Option(name: .customLong("show-watermark"), default: true, help: "Shows/hides the watermark [true | false]")
+    var watermark: Bool
+    
+    var projectURL: URL { URL(fileURLWithPath: project.trimmingEmptyCharacters.expandingTildeInPath) }
+    var outputURL: URL  { URL(fileURLWithPath: output.trimmingEmptyCharacters.expandingTildeInPath) }
+    
+    
+    public func main() -> IO<CLIKit.Console.Error, Void> {
+        arguments(parsableCommand: self)
+            .flatMap { (input, output, style) in
+                nef.Carbon.render(playgroundsAt: input, style: style, into: output)
+                    .provide(Console.default)^
+                    .mapError { _ in .render() }
+                    .foldM({ _ in Console.default.exit(failure: "rendering Xcode Playgrounds from '\(input.path)'") },
+                           { _ in Console.default.exit(success: "rendered Xcode Playgrounds in '\(output.path)'")   })
+            }^
+    }
+    
+    private func arguments(parsableCommand: CarbonCommand) -> IO<CLIKit.Console.Error, (input: URL, output: URL, style: CarbonStyle)> {
```

Comment:
I think working with a tuple were is not very convenient. It should be moved to its own data type.

Reply:
Yep, I think about it, maybe a lazy decision...